File: algorithm/array_and_matrices/242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dictionary = {}
        for i in s:
            if i in dictionary:
                dictionary[i] += 1
            else:
                dictionary[i] = 1
        for j in t:
            if j in dictionary:
                dictionary[j] -= 1
            else:
                return False
        return all(i == 0 for i in dictionary.values())
    
if __name__ == "__main__":
    s = "anagram"
    t = "nagaram"
    print(Solution().isAnagram(s,t))

File: 242.cpp
#include <iostream>
#include <map>
#include <vector>
using namespace std;

class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.length() != t.length())
        {
            return false;
        }
        vector<int> count(26, 0);
        for (int i = 0; i <= s.length() - 1; i++)
        {
            count[s[i] - 'a'] += 1;
            count[t[i] - 'a'] -= 1;
        }
        for (const auto &n : count)
        {
            if (n != 0)
            {
                return false;
            }
        }
        return true;
    }
};

int main()
{
    Solution sol;
    string s = "anagram", t = "nagaram";
    cout << boolalpha << sol.isAnagram(s, t);
}